Islandora-CLAW/Crayfish
# Fixing media indexing in Fedora with 5.0

**GitHub Issue**: https://github.com/Islandora-CLAW/CLAW/issues/1079

Test with: https://github.com/Islandora-Devops/claw-playbook/pull/100

# What does this Pull Request do?

Fixes Milliner indexing of media for files that are in the fedora.

# What's new?
We were indexing the RDF with the subject of the triples being the file uri itself: `http://localhost:8080/fcrepo/rest/some_file.jpg`. Fedora has been rejecting these with 403's because it requires the fcr:medata url to be the subject: `http://localhost:8080/fcrepo/rest/some_file.jpg/fcr:metadata`

This is a one line fix to inject that url in as the subject instead of the file uri.

# How should this be tested?

You should be able to see things like `pcdm:fileOf` and `schema:author` in RDF from Fedora now:

```bash
vagrant@claw:~$ curl http://localhost:8080/fcrepo/rest/2019-04/beefalo.jpeg/fcr:metadata
@prefix premis:  <http://www.loc.gov/premis/rdf/v1#> .
@prefix rdf:  <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix fedora:  <http://fedora.info/definitions/v4/repository#> .
@prefix ebucore:  <http://www.ebu.ch/metadata/ontologies/ebucore/ebucore#> .
@prefix ldp:  <http://www.w3.org/ns/ldp#> .
@prefix dcterms:  <http://purl.org/dc/terms/> .
@prefix iana:  <http://www.iana.org/assignments/relation/> .

<http://localhost:8080/fcrepo/rest/2019-04/beefalo.jpeg>
        rdf:type                    fedora:NonRdfSourceDescription ;
        rdf:type                    <http://pcdm.org/models#File> ;
        rdf:type                    <http://pcdm.org/use#OriginalFile> ;
        rdf:type                    fedora:Binary ;
        rdf:type                    fedora:Resource ;
        fedora:lastModifiedBy       "bypassAdmin" ;
        <http://schema.org/dateModified>  "2019-04-16T13:21:12+00:00"^^<http://www.w3.org/2001/XMLSchema#dateTime> ;
        <http://schema.org/author>  <http://localhost:8000/user/1?_format=jsonld> ;
        <http://schema.org/sameAs>  "http://localhost:8000/_flysystem/fedora/2019-04/beefalo.jpeg" ;
        <http://schema.org/dateCreated>  "2019-04-16T12:55:35+00:00"^^<http://www.w3.org/2001/XMLSchema#dateTime> ;
        ebucore:width               "1024"^^<http://www.w3.org/2001/XMLSchema#int> ;
        premis:hasSize              "194003"^^<http://www.w3.org/2001/XMLSchema#long> ;
        ebucore:hasMimeType         "image/jpeg" ;
        fedora:createdBy            "bypassAdmin" ;
        fedora:created              "2019-04-16T12:55:41.454Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> ;
        fedora:lastModified         "2019-04-16T13:21:13.939Z"^^<http://www.w3.org/2001/XMLSchema#dateTime> ;
        premis:hasMessageDigest     <urn:sha1:5079e6e28144b97e9019d8fa5a831e8049858fcc> ;
        ebucore:height              "576"^^<http://www.w3.org/2001/XMLSchema#int> ;
        <http://pcdm.org/models#fileOf>  <http://localhost:8000/node/2?_format=jsonld> ;
        rdf:label                   "Beefalo Madness"@en ;
        ebucore:filename            "" ;
        dcterms:title               "Beefalo Madness"@en ;
        rdf:type                    ldp:NonRDFSource ;
        iana:describedby            <http://localhost:8080/fcrepo/rest/2019-04/beefalo.jpeg/fcr:metadata> ;
        fedora:hasFixityService     <http://localhost:8080/fcrepo/rest/2019-04/beefalo.jpeg/fcr:fixity> .

```

# Interested parties
@Islandora-CLAW/committers
